dpe/src/commands/sign.rs

```diff
@@ -282,6 +333,45 @@ mod tests {
         assert!(sig.verify(&TEST_DIGEST, &ec_pub_key).unwrap());
     }
 
+    #[test]
+    fn test_symmetric() {
+        let mut dpe = DpeInstance::<OpensslCrypto>::new_for_test(
+            Support {
+                auto_init: true,
+                is_symmetric: true,
+                ..Support::default()
+            },
+            &TEST_LOCALITIES,
+        )
+        .unwrap();
+
+        for i in 0..3 {
+            DeriveChildCmd {
+                handle: ContextHandle::default(),
+                data: [i; DPE_PROFILE.get_hash_size()],
+                flags: DeriveChildCmd::MAKE_DEFAULT,
+                tci_type: i as u32,
+                target_locality: 0,
+            }
+            .execute(&mut dpe, TEST_LOCALITIES[0])
+            .unwrap();
+        }
+
+        let cmd = SignCmd {
+            handle: ContextHandle::default(),
+            label: TEST_LABEL,
+            flags: SignCmd::IS_SYMMETRIC,
+            digest: TEST_DIGEST,
+        };
+        let resp = match cmd.execute(&mut dpe, TEST_LOCALITIES[0]).unwrap() {
+            Response::Sign(resp) => resp,
+            _ => panic!("Incorrect response type"),
+        };
+
+        // Check that we are using HMAC, i.e. s must be a buffer of all 0s
+        assert!(&resp.sig_s.iter().all(|&b| b == 0x0));
```

Comment:
It would be good to verify the HMAC here too. You'll have to re-derive the HMAC.